Second homework

DariaMaslova opened the pull request:
Build a screen used at least those widgets: Container, Center, Text, TextView, Icon, FloatingActionButton, ListView, Row, ListTile, EdgeInsets, BoxDecoration and more.

Commit:
Second homework

Build a screen used at least those widgets: Container, Center, Text, TextView, Icon, FloatingActionButton, ListView, Row, ListTile, EdgeInsets, BoxDecoration and more.

GamaGong commented on lib/pages/myplants/mock.dart:
```diff
@@ -0,0 +1,42 @@
+
+import 'model/plant.dart';
+
+final _plants = [
```
Можно использовать ```const``` https://dart.dev/guides/language/language-tour#final-and-const

DariaMaslova replied:
Добавила

GamaGong commented on lib/pages/myplants/my_plants_page.dart:
```diff
@@ -0,0 +1,324 @@
+      color: Colors.white,
+    );
+    switch (plant.hydrationState) {
+      case HydrationState.Dry:
```
Похоже что можно создать виджет, который будет отображать ```HydrationState``` и\или создать все вариации  ```HydrationState``` и пометить их const, чтобы не создавать одно и то же много раз.

DariaMaslova replied:
Вынесла в отдельный виджет ```HydrationStatus```

antonBatrakov commented on lib/pages/myplants/my_plants_page.dart:
```diff
@@ -0,0 +1,324 @@
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          "My Plants",
```
Можно заранее начать выносить стринговые константы в отдельный файл, чтобы потом с локализацией не мучаться, когда всё разрастется.

DariaMaslova replied:
Вынесла

antonBatrakov commented on lib/pages/myplants/my_plants_page.dart:
```diff
@@ -0,0 +1,324 @@
+            flex: 7,
+            child: Container(
+              color: Colors.white,
+              child: ListView(
```
А почему бы не использовать Drawer вместо Row -> [Expanded, Spacer] -> Container ?

DariaMaslova replied:
Заменила на ```Drawer```

GamaGong commented on lib/pages/myplants/my_plants_page.dart:
```diff
@@ -0,0 +1,324 @@
+                    ),
+                    decoration: ShapeDecoration(
+                      color: Colors.green,
+                      shape: LeafBottomShapeBorder(),
```
Кстати, тут у тебя константный конструктор создал не константный объект. нужен const

DariaMaslova replied:
Не совсем поняла, что имеется ввиду.

1. У ```ShapeDecoration```'а выше по дереву вроде нигде константный конструктор не используется
2. Даже если бы использовался где-то, то все параметры по умолчанию создавались с [констатными конструкторами](https://dart.dev/guides/language/effective-dart/usage#dont-use-const-redundantly)
3. Если бы в случае (2) у ```ShapeDecoration```'а использовался бы не константый конструктор, то билдбы не собрался и студия десять раз подчеркнула ошибку

Но я все равно добавила ```const```

Commit:
fixes comments

DenisovAV commented on lib/pages/common/models/menu.dart:
```diff
@@ -0,0 +1,50 @@
+
+  const Menu({this.items, this.selected});
+
+  factory Menu.appMenu({String selected = MenuTitles.MY_PLANTS,}) {
```
named constructor would be better here

DariaMaslova replied:
changed to named constructor

DenisovAV commented on lib/pages/myplants/my_plants_page.dart:
```diff
@@ -183,11 +93,11 @@ class _PlantItem extends StatelessWidget {
     }
   }
 
-  MainAxisAlignment _waterDropsAlignment() {
```
You can use the conditional operator here, return isLeft?Alignment.centerLeft:Alignment.centerRight;

DariaMaslova replied:
fixed

DenisovAV commented on lib/widgets/hydration_state.dart:
```diff
@@ -0,0 +1,67 @@
+      MaterialCommunityIcons.water_outline,
+      color: color,
+    );
+    switch (hydration) {
```
Can be more short and algorithmic, something like this:
    final dryState = <Widget>[];
    for (var i=3; i>0; i--){
      hydration.index < i ? dryState.add(emptyIcon) : dryState.add(filledIcon);
    }
    return Row(
      mainAxisSize: MainAxisSize.min,
      children: dryState,
    );

DariaMaslova replied:
fixed

Commit:
fixes comments #2

- named constauctor for menu
- conditional oprators for alignments
- algorithmic approach for hydration state
